Autopilot/AttitudeManager/VN_user.c

```diff
@@ -77,74 +77,22 @@ void VN_SPI_SetSS(unsigned char sensorID, VN_PinState state){
 * Return         : The data received on the SPI bus
 *******************************************************************************/
 unsigned long VN_SPI_SendReceive(unsigned long data){
-
 /* User code to send out 4 bytes over SPI goes here */
-  unsigned long ret = 0;
-    /* Wait for SPI1 Tx buffer empty */
-    //while (SPI_I2S_GetFlagStatus(SPI1, SPI_I2S_FLAG_TXE) == RESET);
-  	while (SPI2STATbits.SPITBF){//if (smAllowed){StateMachine(STATEMACHINE_IMU);}
-        }
-    /* Send SPI1 requests */
-    //SPI_I2S_SendData(SPI1, VN_BYTE(data, i));
-	SPI2BUF = VN_BYTE4(data);
-        while (SPI2STATbits.SPITBF){
-        }
-    /* Wait for response from VN-100 */
-    //while (SPI_I2S_GetFlagStatus(SPI1, SPI_I2S_FLAG_RXNE) == RESET);
-  	while (!SPI2STATbits.SPIRBF){//if (smAllowed){StateMachine(STATEMACHINE_IMU);}
-        }
-        
-    /* Save received data in buffer */
-    //ret |= ((unsigned long)SPI_I2S_ReceiveData(SPI1) << (8*i));
-	ret |= ((unsigned long)SPI2BUF << (0));
-
-  	while (SPI2STATbits.SPITBF){
-        }
-    /* Send SPI1 requests */
-    //SPI_I2S_SendData(SPI1, VN_BYTE(data, i));
-	SPI2BUF = VN_BYTE3(data);
-
-    /* Wait for response from VN-100 */
-    //while (SPI_I2S_GetFlagStatus(SPI1, SPI_I2S_FLAG_RXNE) == RESET);
-  	while (!SPI2STATbits.SPIRBF){
-        }
-    /* Save received data in buffer */
-    //ret |= ((unsigned long)SPI_I2S_ReceiveData(SPI1) << (8*i));
-	ret |= ((unsigned long)SPI2BUF << (8));
-
-  	while (SPI2STATbits.SPITBF){
-        }
-    /* Send SPI1 requests */
-    //SPI_I2S_SendData(SPI1, VN_BYTE(data, i));
-	SPI2BUF = VN_BYTE2(data);
-
-    /* Wait for response from VN-100 */
-    //while (SPI_I2S_GetFlagStatus(SPI1, SPI_I2S_FLAG_RXNE) == RESET);
-  	while (!SPI2STATbits.SPIRBF){
-        }
-
-    /* Save received data in buffer */
-    //ret |= ((unsigned long)SPI_I2S_ReceiveData(SPI1) << (8*i));
-	ret |= ((unsigned long)SPI2BUF << (16));
-
-  	while (SPI2STATbits.SPITBF){
-        }
-
-    /* Send SPI1 requests */
-    //SPI_I2S_SendData(SPI1, VN_BYTE(data, i));
-	SPI2BUF = VN_BYTE1(data);
-
-    /* Wait for response from VN-100 */
-    //while (SPI_I2S_GetFlagStatus(SPI1, SPI_I2S_FLAG_RXNE) == RESET);
-  	while (!SPI2STATbits.SPIRBF){
-        }
-    /* Save received data in buffer */
-    //ret |= ((unsigned long)SPI_I2S_ReceiveData(SPI1) << (8*i));
-	ret |= ((unsigned long)SPI2BUF << (24));
-
-
-
-  return ret;
+    unsigned long ret = 0;
+    
+    /* Wait for SPI2 Tx buffer empty */
+  	while (SPI2STATbits.SPITBF){}
```

Comment:
I'm not certain. It was there before, but I don't think it's strictly necessary.